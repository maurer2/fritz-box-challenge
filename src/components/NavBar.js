import React from 'react';
import styled from 'styled-components/macro';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';

import NavBarEntry from './NavBarEntry';

const BoxInformationWrapper = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  justify-content: space-between;
  flex-wrap: wrap;
  background: red;
  list-style: none;
`;

const SlideYTransition = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;

  .slide-appear,
  .slide-enter {
    transform: translateY(100%);
  }

  .slide-appear.slide-appear-active,
  .slide-enter.slide-enter-active {
    transform: translateY(0);
    transition: transform 250ms ease-out;
  }

  .slide-leave {
    transform: translateY(0);
  }

  .slide-leave.slide-leave-active {
    transform: translateY(100%);
    transition: transform 250ms ease-out;
  }
`;

const BoxInformationComponent = (props) => {
  const { boxData, isUpdating, currentIndex, handleNavigation } = props;

  return (
    <SlideYTransition>
      <CSSTransitionGroup
        component={ React.Fragment }
        transitionAppear={ false }
        transitionName="slide"
        transitionAppearTimeout={ 0 }
        transitionLeaveTimeout={ 250 }
        transitionEnterTimeout={ 250 }
      >
        { !isUpdating && (
          <BoxInformationWrapper>
            { Object.keys(boxData).map((entry, index) => (
              <NavBarEntry
                index={ index }
                entry={ entry }
                isActive={ currentIndex === index }
                key={ index }
                handleNavigation={ handleNavigation }
              />
            ))}
          </BoxInformationWrapper>
        )}
      </CSSTransitionGroup>
    </SlideYTransition>
  );
};

BoxInformationComponent.propTypes = {
  boxData: PropTypes.object,
  isUpdating: PropTypes.bool,
  handleNavigation: PropTypes.func,
  currentIndex: PropTypes.number,
};

export default BoxInformationComponent;
