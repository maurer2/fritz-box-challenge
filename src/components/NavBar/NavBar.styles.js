import styled from 'styled-components/macro';

export const NavBar = styled.div`
  grid-area: navBar;
  position: relative;
`;

export const SlideYTransition = styled.div`
  > .${(props) => props.transitionName}-appear {
    transform: translateY(100%);
    transition: transform 500ms ease-out;

    &.${(props) => props.transitionName}-appear-active {
      transform: translateY(0);
    }
  }

  > .${(props) => props.transitionName}-leave {
    transform: translateY(0);
    transition: transform 500ms ease-in;

    &.${(props) => props.transitionName}-leave-active {
      transform: translateY(100%);
    }
  }
`;

export const NavBarWrapper = styled.footer`
  position: relative;
  padding-top: ${({ reservedSpaceTop }) => reservedSpaceTop}px;
`;

export const Indicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: ${(props) => (props.width === 'auto' ? props.width : `${props.width}px`)}; // might cause layout thrashing
  height: ${(props) => props.height}px;
  transform: translateX(${(props) => props.offset}px);
  transition: transform 500ms, width 500ms;
  background: white;
`;

export const NavBarList = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  flex-wrap: ${(props) => (props.isRow ? 'no-wrap' : 'wrap')};
  justify-content: space-between;
  list-style: none;
  background: #BDBDBD;
`;
