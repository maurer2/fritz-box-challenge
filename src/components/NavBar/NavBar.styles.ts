import styled, { css } from 'styled-components/macro';

export const NavBar = styled.nav`
  position: relative;
  padding-top: ${({ reservedSpaceTop }) => `${reservedSpaceTop}px`};
  grid-area: navBar;
  will-change: transform;
  transform: translateY(0);
  transition: transform 500ms ease-out;

  ${({ isUpdating }) =>
    isUpdating &&
    css`
      transform: translateY(100%);
    `}
`;

export const Indicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: ${({ width }) =>
    width === 'auto' ? width : `${width}px`}; // might cause layout thrashing
  height: ${({ height }) => `${height}px`};
  transform: translateX(${({ offset }) => offset}px);
  transition: transform 500ms, width 500ms;
  background: ${({ theme }) => theme.primaryColor};
`;

export const NavBarList = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  flex-wrap: ${({ isRow }) => (isRow ? 'no-wrap' : 'wrap')};
  justify-content: space-between;
  list-style: none;
  background: ${({ theme }) => theme.secondaryColor};
`;
