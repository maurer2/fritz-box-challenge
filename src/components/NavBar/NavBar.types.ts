export interface StringMap {
  [key: string]: string;
}

export interface NavBarProps {
  componentsToShow: any[];
  currentIndex: number;
  handleNavigation: any;
  isUpdating: boolean;
}
