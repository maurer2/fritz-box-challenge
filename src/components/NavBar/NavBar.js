import React, { useState, useRef, useEffect, createRef } from 'react';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';

// import { throttle } from 'lodash';
import * as Styles from './NavBar.styles';

import { NavBarEntry } from '../NavBarEntry';

const Wrapper = ({ children }) => {
  const transitionName = 'slide-vertically';

  return (
    <Styles.SlideYTransition transitionName={transitionName}>
      <CSSTransitionGroup
        component={React.Fragment}
        transitionAppear={false}
        transitionEnter={false}
        transitionName={transitionName}
        transitionAppearTimeout={500}
        transitionLeaveTimeout={500}
        transitionEnterTimeout={0}
      >
        { children }
      </CSSTransitionGroup>
    </Styles.SlideYTransition>
  );
};

const NavBar = ({ componentsToShow, currentIndex, handleNavigation }) => {
  // const trottledResizeHandler = useRef({});
  const [offset, setOffset] = useState(0);
  const [width, setWidth] = useState('auto'); // prevent css transition on load

  const oldIndex = useRef(-1);
  const showIndicator = useRef(true);
  const activeElement = createRef();

  const height = 5;

  useEffect(() => {
    // trottledResizeHandler.current = throttle(handleResize, 300);
    const activeElementHasChanged = currentIndex !== oldIndex.current;

    if (activeElementHasChanged) {
      oldIndex.current = currentIndex;

      updateIndicator();
    }

    /*
    return () => {
    };
    */
  });

  /*
  function handleResize() {
    updateIndicator();
  }
  */

  function updateIndicator() {
    if (activeElement.current == null) {
      return;
    }

    const activeElementBB = activeElement.current.getBoundingClientRect();
    const isLargeScreen = window.matchMedia('(min-width: 768px)').matches;

    setOffset(activeElementBB.x);
    setWidth(activeElementBB.width);

    showIndicator.current = isLargeScreen;
  }

  return (
    <Wrapper>
      <Styles.NavBarWrapper reservedSpaceTop={height}>
        { showIndicator && (
        <Styles.Indicator
          offset={offset}
          width={width}
          height={height}
        />
        )}
        <Styles.NavBarList isRow={showIndicator}>
          { componentsToShow.map((entry, index) => (
            <NavBarEntry
              index={index}
              entry={entry}
              isActive={currentIndex === index}
              handleNavigation={handleNavigation}
              // only active element gets ref otherwise last child always active
              activeElementRef={(currentIndex === index) ? activeElement : null}
              key={index}
              isFullWidth={!showIndicator}
            />
          ))}
        </Styles.NavBarList>
      </Styles.NavBarWrapper>
    </Wrapper>
  );
};

const { string, number, func, node } = PropTypes;

NavBar.propTypes = {
  componentsToShow: PropTypes.arrayOf(string).isRequired,
  handleNavigation: func.isRequired,
  currentIndex: number.isRequired,
};

Wrapper.propTypes = {
  children: node.isRequired,
};

export { NavBar };
