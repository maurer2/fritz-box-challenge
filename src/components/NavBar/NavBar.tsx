import React, { useState, useRef, useEffect, createRef } from 'react';
import PropTypes from 'prop-types';

// import { throttle } from 'lodash';
import * as Styles from './NavBar.styles';
import * as Types from './NavBar.types';

import { NavBarEntry } from '../NavBarEntry';

const NavBar: React.FC<Types.NavBarProps> = ({ componentsToShow, currentIndex, handleNavigation, isUpdating }) => {
  // const throttledResizeHandler = useRef({});
  const [offset, setOffset] = useState(0);
  const [width, setWidth] = useState('auto'); // prevent css transition on load

  const oldIndex = useRef(-1);
  const showIndicator = useRef(true);
  const activeElement = createRef();

  const height = 5;

  function updateIndicator() {
    if (activeElement.current == null) {
      return;
    }

    const activeElementBB = activeElement.current.getBoundingClientRect();
    const isLargeScreen = window.matchMedia('(min-width: 768px)').matches;

    setOffset(activeElementBB.x);
    setWidth(activeElementBB.width);

    showIndicator.current = isLargeScreen;
  }

  useEffect(() => {
    // throttledResizeHandler.current = throttle(handleResize, 300);
    const activeElementHasChanged = currentIndex !== oldIndex.current;

    if (activeElementHasChanged) {
      oldIndex.current = currentIndex;

      updateIndicator();
    }
  });

  return (
    <Styles.NavBar reservedSpaceTop={height} isUpdating={isUpdating}>
      { showIndicator.current && (
      <Styles.Indicator
        offset={offset}
        width={width}
        height={height}
      />
      )}
      <Styles.NavBarList isRow={showIndicator}>
        { componentsToShow.map((entry, index) => (
          <NavBarEntry
            index={index}
            entry={entry}
            isActive={currentIndex === index}
            handleNavigation={handleNavigation}
            // only active element gets ref otherwise last child always active
            activeElementRef={(currentIndex === index) ? activeElement : null}
            key={index}
            isFullWidth={!showIndicator}
          />
        ))}
      </Styles.NavBarList>
    </Styles.NavBar>
  );
};

const { string, number, func, bool } = PropTypes;

NavBar.propTypes = {
  componentsToShow: PropTypes.arrayOf(string).isRequired,
  handleNavigation: func.isRequired,
  currentIndex: number.isRequired,
  isUpdating: bool.isRequired,
};

export { NavBar };
