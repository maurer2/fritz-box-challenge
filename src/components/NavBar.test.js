import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

import NavBar from './NavBar';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('NavBar', () => {
  const boxData = {
    branding: 'avm',
    model: 'FRITZ!Box 6590 Cable',
    restarts: '024',
    technology: 'Kabel',
  };

  const mockedHandleNavigation = jest.fn();

  const wrapper = shallow(<NavBar
    isUpdating={ false }
    boxData={ boxData }
    currentIndex={ 1 }
    handleNavigation={ mockedHandleNavigation }
  />);

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a BoxInformationWrapper component', () => {
    expect(wrapper.find('NavBarWrapper').length).toBe(1);
  });

  it('should have children', () => {
    expect(wrapper.find('NavBarWrapper').children().length).toBeGreaterThan(0);
  });
});
