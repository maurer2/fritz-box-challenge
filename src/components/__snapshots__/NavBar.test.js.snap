// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    boxData={
      Object {
        "branding": "avm",
        "model": "FRITZ!Box 6590 Cable",
        "restarts": "024",
        "technology": "Kabel",
      }
    }
    currentIndex={1}
    handleNavigation={[MockFunction]}
    isUpdating={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CSSTransitionGroup
        component={Symbol(react.fragment)}
        transitionAppear={true}
        transitionAppearTimeout={500}
        transitionEnter={false}
        transitionEnterTimeout={0}
        transitionLeave={true}
        transitionLeaveTimeout={500}
        transitionName="slide-vertically"
      >
        <ForwardRef
          reservedSpaceTop={5}
        >
          <ForwardRef
            height={5}
            offset={0}
            width="auto"
          />
          <ForwardRef>
            <NavBarEntry
              activeElementRef={null}
              entry="branding"
              handleNavigation={[MockFunction]}
              index={0}
              isActive={false}
            />
            <NavBarEntry
              activeElementRef={
                Object {
                  "current": null,
                }
              }
              entry="model"
              handleNavigation={[MockFunction]}
              index={1}
              isActive={true}
            />
            <NavBarEntry
              activeElementRef={null}
              entry="restarts"
              handleNavigation={[MockFunction]}
              index={2}
              isActive={false}
            />
            <NavBarEntry
              activeElementRef={null}
              entry="technology"
              handleNavigation={[MockFunction]}
              index={3}
              isActive={false}
            />
          </ForwardRef>
        </ForwardRef>
      </CSSTransitionGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef
          reservedSpaceTop={5}
        >
          <ForwardRef
            height={5}
            offset={0}
            width="auto"
          />
          <ForwardRef>
            <NavBarEntry
              activeElementRef={null}
              entry="branding"
              handleNavigation={[MockFunction]}
              index={0}
              isActive={false}
            />
            <NavBarEntry
              activeElementRef={
                Object {
                  "current": null,
                }
              }
              entry="model"
              handleNavigation={[MockFunction]}
              index={1}
              isActive={true}
            />
            <NavBarEntry
              activeElementRef={null}
              entry="restarts"
              handleNavigation={[MockFunction]}
              index={2}
              isActive={false}
            />
            <NavBarEntry
              activeElementRef={null}
              entry="technology"
              handleNavigation={[MockFunction]}
              index={3}
              isActive={false}
            />
          </ForwardRef>
        </ForwardRef>,
        "component": Symbol(react.fragment),
        "transitionAppear": true,
        "transitionAppearTimeout": 500,
        "transitionEnter": false,
        "transitionEnterTimeout": 0,
        "transitionLeave": true,
        "transitionLeaveTimeout": 500,
        "transitionName": "slide-vertically",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              height={5}
              offset={0}
              width="auto"
            />,
            <ForwardRef>
              <NavBarEntry
                activeElementRef={null}
                entry="branding"
                handleNavigation={[MockFunction]}
                index={0}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={
                  Object {
                    "current": null,
                  }
                }
                entry="model"
                handleNavigation={[MockFunction]}
                index={1}
                isActive={true}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="restarts"
                handleNavigation={[MockFunction]}
                index={2}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="technology"
                handleNavigation={[MockFunction]}
                index={3}
                isActive={false}
              />
            </ForwardRef>,
          ],
          "reservedSpaceTop": 5,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 5,
              "offset": 0,
              "width": "auto",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Indicator-sc-13xfpz1-2",
                "isStatic": false,
                "rules": Array [
                  "position:absolute;top:0;left:0;width:",
                  [Function],
                  ";height:",
                  [Function],
                  "px;transform:translateX(",
                  [Function],
                  "px);transition:transform 500ms,width 500ms;background:white;",
                ],
              },
              "displayName": "Indicator",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Indicator-sc-13xfpz1-2",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavBarEntry
                  activeElementRef={null}
                  entry="branding"
                  handleNavigation={[MockFunction]}
                  index={0}
                  isActive={false}
                />,
                <NavBarEntry
                  activeElementRef={
                    Object {
                      "current": null,
                    }
                  }
                  entry="model"
                  handleNavigation={[MockFunction]}
                  index={1}
                  isActive={true}
                />,
                <NavBarEntry
                  activeElementRef={null}
                  entry="restarts"
                  handleNavigation={[MockFunction]}
                  index={2}
                  isActive={false}
                />,
                <NavBarEntry
                  activeElementRef={null}
                  entry="technology"
                  handleNavigation={[MockFunction]}
                  index={3}
                  isActive={false}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "activeElementRef": null,
                  "entry": "branding",
                  "handleNavigation": [MockFunction],
                  "index": 0,
                  "isActive": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "activeElementRef": Object {
                    "current": null,
                  },
                  "entry": "model",
                  "handleNavigation": [MockFunction],
                  "index": 1,
                  "isActive": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "class",
                "props": Object {
                  "activeElementRef": null,
                  "entry": "restarts",
                  "handleNavigation": [MockFunction],
                  "index": 2,
                  "isActive": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "class",
                "props": Object {
                  "activeElementRef": null,
                  "entry": "technology",
                  "handleNavigation": [MockFunction],
                  "index": 3,
                  "isActive": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "NavBarList-sc-13xfpz1-3",
                "isStatic": true,
                "rules": Array [
                  "display:flex;margin:0;padding:0;justify-content:space-between;flex-wrap:wrap;list-style:none;background:red;",
                ],
              },
              "displayName": "NavBarList",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "NavBarList-sc-13xfpz1-3",
              "target": "ul",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "NavBarWrapper-sc-13xfpz1-1",
            "isStatic": false,
            "rules": Array [
              "position:relative;display:block;padding-top:",
              [Function],
              "px;",
            ],
          },
          "displayName": "NavBarWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "NavBarWrapper-sc-13xfpz1-1",
          "target": "footer",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SlideYTransition-sc-13xfpz1-0",
        "isStatic": true,
        "rules": Array [
          "position:absolute;left:0;right:0;bottom:0;> .slide-vertically-appear{transform:translateY(100%);&.slide-vertically-appear-active{transform:translateY(0);transition:transform 500ms ease-out;}}> .slide-vertically-leave{transform:translateY(0);&.slide-vertically-leave-active{transform:translateY(100%);transition:transform 500ms ease-in;}}",
        ],
      },
      "displayName": "SlideYTransition",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SlideYTransition-sc-13xfpz1-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CSSTransitionGroup
          component={Symbol(react.fragment)}
          transitionAppear={true}
          transitionAppearTimeout={500}
          transitionEnter={false}
          transitionEnterTimeout={0}
          transitionLeave={true}
          transitionLeaveTimeout={500}
          transitionName="slide-vertically"
        >
          <ForwardRef
            reservedSpaceTop={5}
          >
            <ForwardRef
              height={5}
              offset={0}
              width="auto"
            />
            <ForwardRef>
              <NavBarEntry
                activeElementRef={null}
                entry="branding"
                handleNavigation={[MockFunction]}
                index={0}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={
                  Object {
                    "current": null,
                  }
                }
                entry="model"
                handleNavigation={[MockFunction]}
                index={1}
                isActive={true}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="restarts"
                handleNavigation={[MockFunction]}
                index={2}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="technology"
                handleNavigation={[MockFunction]}
                index={3}
                isActive={false}
              />
            </ForwardRef>
          </ForwardRef>
        </CSSTransitionGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef
            reservedSpaceTop={5}
          >
            <ForwardRef
              height={5}
              offset={0}
              width="auto"
            />
            <ForwardRef>
              <NavBarEntry
                activeElementRef={null}
                entry="branding"
                handleNavigation={[MockFunction]}
                index={0}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={
                  Object {
                    "current": null,
                  }
                }
                entry="model"
                handleNavigation={[MockFunction]}
                index={1}
                isActive={true}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="restarts"
                handleNavigation={[MockFunction]}
                index={2}
                isActive={false}
              />
              <NavBarEntry
                activeElementRef={null}
                entry="technology"
                handleNavigation={[MockFunction]}
                index={3}
                isActive={false}
              />
            </ForwardRef>
          </ForwardRef>,
          "component": Symbol(react.fragment),
          "transitionAppear": true,
          "transitionAppearTimeout": 500,
          "transitionEnter": false,
          "transitionEnterTimeout": 0,
          "transitionLeave": true,
          "transitionLeaveTimeout": 500,
          "transitionName": "slide-vertically",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                height={5}
                offset={0}
                width="auto"
              />,
              <ForwardRef>
                <NavBarEntry
                  activeElementRef={null}
                  entry="branding"
                  handleNavigation={[MockFunction]}
                  index={0}
                  isActive={false}
                />
                <NavBarEntry
                  activeElementRef={
                    Object {
                      "current": null,
                    }
                  }
                  entry="model"
                  handleNavigation={[MockFunction]}
                  index={1}
                  isActive={true}
                />
                <NavBarEntry
                  activeElementRef={null}
                  entry="restarts"
                  handleNavigation={[MockFunction]}
                  index={2}
                  isActive={false}
                />
                <NavBarEntry
                  activeElementRef={null}
                  entry="technology"
                  handleNavigation={[MockFunction]}
                  index={3}
                  isActive={false}
                />
              </ForwardRef>,
            ],
            "reservedSpaceTop": 5,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": 5,
                "offset": 0,
                "width": "auto",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Indicator-sc-13xfpz1-2",
                  "isStatic": false,
                  "rules": Array [
                    "position:absolute;top:0;left:0;width:",
                    [Function],
                    ";height:",
                    [Function],
                    "px;transform:translateX(",
                    [Function],
                    "px);transition:transform 500ms,width 500ms;background:white;",
                  ],
                },
                "displayName": "Indicator",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Indicator-sc-13xfpz1-2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavBarEntry
                    activeElementRef={null}
                    entry="branding"
                    handleNavigation={[MockFunction]}
                    index={0}
                    isActive={false}
                  />,
                  <NavBarEntry
                    activeElementRef={
                      Object {
                        "current": null,
                      }
                    }
                    entry="model"
                    handleNavigation={[MockFunction]}
                    index={1}
                    isActive={true}
                  />,
                  <NavBarEntry
                    activeElementRef={null}
                    entry="restarts"
                    handleNavigation={[MockFunction]}
                    index={2}
                    isActive={false}
                  />,
                  <NavBarEntry
                    activeElementRef={null}
                    entry="technology"
                    handleNavigation={[MockFunction]}
                    index={3}
                    isActive={false}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "class",
                  "props": Object {
                    "activeElementRef": null,
                    "entry": "branding",
                    "handleNavigation": [MockFunction],
                    "index": 0,
                    "isActive": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "activeElementRef": Object {
                      "current": null,
                    },
                    "entry": "model",
                    "handleNavigation": [MockFunction],
                    "index": 1,
                    "isActive": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "class",
                  "props": Object {
                    "activeElementRef": null,
                    "entry": "restarts",
                    "handleNavigation": [MockFunction],
                    "index": 2,
                    "isActive": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "class",
                  "props": Object {
                    "activeElementRef": null,
                    "entry": "technology",
                    "handleNavigation": [MockFunction],
                    "index": 3,
                    "isActive": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "NavBarList-sc-13xfpz1-3",
                  "isStatic": true,
                  "rules": Array [
                    "display:flex;margin:0;padding:0;justify-content:space-between;flex-wrap:wrap;list-style:none;background:red;",
                  ],
                },
                "displayName": "NavBarList",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "NavBarList-sc-13xfpz1-3",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "NavBarWrapper-sc-13xfpz1-1",
              "isStatic": false,
              "rules": Array [
                "position:relative;display:block;padding-top:",
                [Function],
                "px;",
              ],
            },
            "displayName": "NavBarWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "NavBarWrapper-sc-13xfpz1-1",
            "target": "footer",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SlideYTransition-sc-13xfpz1-0",
          "isStatic": true,
          "rules": Array [
            "position:absolute;left:0;right:0;bottom:0;> .slide-vertically-appear{transform:translateY(100%);&.slide-vertically-appear-active{transform:translateY(0);transition:transform 500ms ease-out;}}> .slide-vertically-leave{transform:translateY(0);&.slide-vertically-leave-active{transform:translateY(100%);transition:transform 500ms ease-in;}}",
          ],
        },
        "displayName": "SlideYTransition",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SlideYTransition-sc-13xfpz1-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
